Modelica/Math/isPowerOf2.mo

```diff
@@ -3,9 +3,23 @@ function isPowerOf2 "Determine if the integer input is a power of 2"
   extends Modelica.Icons.Function;
   input Integer i(min=1) "Integer scalar";
   output Boolean result "= true, if integer scalar is a power of 2";
+protected 
+  Integer target;
+  Integer powOf2;
 algorithm
   assert(i>=1, "Integer input to isPowerOf2 has to be >= 1");
-  result := i == 2^integer(log(i)/log(2)+0.5);
+  /* To avoid overflow we divide input by 2, but first check whether it is odd, with special case for 1. */
+  /* And then we check whether 2^... equals the target. */
+  if mod(i, 2) == 1 then
+    result := i == 1;
+  else
+    target := div(i, 2);
+    powOf2 := 1;
+    while powOf2 < target loop
+      powOf2 := powOf2 * 2;
+    end while;
+    result := target == powOf2;
+  end if;
   annotation (Inline=true, Documentation(info="<html>
```

Comment:
>> Does Inline = true still really make sense?
>
>As far as I can see as much sense as before.

I though the degree of making sense would depend on how complex the body of the function is.  The old one just had an assert and and expression giving the result.  Now, the expression for the result is replaced by a much more complex algorithm with both and `if`-statement and a `while`-loop.  To me, having a function call would seem preferable compared to an inlined function body with protected helper variable and all.

Reply:
I just removed `Inline=true` now. I don't see that inlining this function gives any advantage in relevant use cases.

In C it would be different, but the normal C-implementation would be something like `i==1 || !(i & (i-1))`.

Reply:
Yeah, for someone coming from C this must look completely insane – but probably a little bit less insane after removing `Inline = true`.  Resolving.